pymeasure/instruments/hp/hplegacyinstrument.py

```diff
@@ -0,0 +1,218 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import ctypes
+import logging
+import struct
+from pymeasure.instruments import Instrument
+
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+c_uint8 = ctypes.c_uint8
+c_uint16 = ctypes.c_uint16
+c_uint32 = ctypes.c_uint32
+
+
+class HPLegacyInstrument(Instrument):
+    """
+    Class for legacy HP instruments from the era before SPCI, based on `pymeasure.Instrument`
+
+    """
+
+    def __init__(self, adapter, name, status_bytes, status_bitfield, **kwargs):
+        super().__init__(
+            adapter, name,
+            includeSCPI=False,
+            **kwargs,
+        )
+
+        self.name = name
+        self.adapter = adapter
+        self.status_bytes_count = status_bytes
+        self.status_bits = status_bitfield
+        self.status_bytes = self.bytefield_factory(status_bytes)
+        self.status_union = self.union_factory(self.status_bytes, self.status_bits)
+
+        log.info(f"Initializing {self.name}")
+
+    def write(self, command):
+        if command == "B":
+            self.adapter.connection.write("B", termination="")
+        else:
+            # super().write(command)
+            self.adapter.connection.write(command)
+
+    def values(self, command, **kwargs):
+        if command == "B":
+            self.write(command)
+            return self.adapter.connection.read_raw(**kwargs)
+        else:
+            return super().values(command, **kwargs)
+
+    @property
+    def status(self):
+        """
+        Returns an object representing the current status of the unit.
+
+        """
+        current_status = self.decode_status(self, self.fetch_status())
+        return current_status
+
+    def fetch_status(self):
+        """Method to read the status bytes from the instrument
+        :return current_status: a byte array representing the instrument status
+        :rtype current_status: bytes
+        """
+        self.write("B")
+        current_status = self.adapter.read_bytes(self.status_bytes_count)
+        return current_status
+
+    @staticmethod
+    def decode_status(self, status_bytes, field=None):
```

Comment:
This is a bit weird: a static method requiring the instance parameter.
I would say that you can change it in a regular method